internal/fnruntime/utils.go

```diff
@@ -46,7 +47,11 @@ func SaveResults(resultsDir string, fnResults *fnresult.ResultList) (string, err
 		return "", err
 	}
 
-	err = ioutil.WriteFile(filePath, out.Bytes(), 0744)
+	if fsys == nil {
```

Comment:
We should make fsys mandatory dependency otherwise it will lead to mix-mode in the code which will be very difficult to reason with. And to signal that, it will be good to take `fsys` as the first argument in all the methods instead of last.

Reply:
Done. Removed mix-mode.